src/main/java/emily/command/Storage.java

```diff
@@ -0,0 +1,121 @@
+package main.java.emily.command;
+
+import main.java.emily.task.Deadline;
+import main.java.emily.task.Event;
+import main.java.emily.task.Task;
+import main.java.emily.task.ToDos;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the txt file and saving tasks in the file
+ */
+public class Storage {
+    private static String filepath;
+
+    Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Returns a list of tasks
+     * Reads the information from the txt file and process into a list of task
+     *
+     * @return an Arraylist of Task
+     * @throws DukeException when the file is invalid
+     */
+    public ArrayList<Task> readData() throws DukeException {
+
+        ArrayList<Task> store = new ArrayList<>();
+
+        File f = new File(filepath);
+        f.getParentFile().mkdirs();
+        try {
+            if (f.exists()) {
+
+                Scanner sc = new Scanner(f);
+                while (sc.hasNext()) {
+                    String line = sc.nextLine();
+                    char type = line.charAt(0);
+                    boolean isCompleted = line.charAt(1) == '1';
+                    String[] temp;
+
+                    switch (type) {
+                        case 'T':
```

Comment:
I think the indentation level of `case` should be inline with that of the `switch` statement?

Reply:
Okay, I have corrected the indentations. Thanks for pointing them out!